go/vt/vttablet/tabletmanager/action_agent.go

```diff
@@ -849,3 +870,17 @@ func (agent *ActionAgent) withRetry(ctx context.Context, description string, wor
 		}
 	}
 }
+
+func validateDemoteMasterType(tabletTypeParam string) (topodatapb.TabletType, error) {
+	tabletType, err := topoproto.ParseTabletType(tabletTypeParam)
+	if err != nil {
+		return topodatapb.TabletType_UNKNOWN, err
+	}
+	if tabletType != topodatapb.TabletType_SPARE && tabletType != topodatapb.TabletType_REPLICA {
+		return topodatapb.TabletType_UNKNOWN, fmt.Errorf("invalid demote_master_type %v; can only be REPLICA or SPARE", tabletType)
+	}
+	if tabletType == topodatapb.TabletType_SPARE && !*mysqlctl.DisableActiveReparents {
```

Comment:
I think you should check for any value that is not REPLICA. I.e. you should not able to use this flag at all unless you also set `--disable_active_reparents`

Reply:
Can you check if it has been passed at all in the Golang flag API?

Reply:
Done. Refactored to parse flag inside validation